homeassistant/components/sonos/speaker.py

```diff
@@ -819,7 +818,7 @@ def _async_regroup(group: list[str]) -> None:
 
             entity_registry = ent_reg.async_get(self.hass)
             sonos_group = []
-            sonos_group_entities = []
+            sonos_group_entities: list = []
```

Comment:
That just defers the issue. I would suggest casting `entity_id` to `str` instead as it shouldn't ever be `None`.